storage/modules/block_storage.go

```diff
@@ -1084,6 +1126,76 @@ func (b *BlockStorage) FindTransaction(
 	return newestBlock, newestTransaction, nil
 }
 
+func (b *BlockStorage) FindRelatedTransactions(
+	ctx context.Context,
+	transactionIdentifier *types.TransactionIdentifier,
+	db database.Transaction,
+) (*types.BlockIdentifier, *types.Transaction, []*types.TransactionIdentifier, error) {
+	maxBlock, tx, err := b.FindTransaction(ctx, transactionIdentifier, db)
+
+	children, err := b.getChildren(ctx, tx, db)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	for _, child := range children {
+		childBlock, childTx, err := b.FindTransaction(ctx, child, db)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+
+		if maxBlock.Index < childBlock.Index {
+			maxBlock = childBlock
+		}
+
+		if childTx == nil {
+			return nil, nil, nil, nil
+		}
+
+		newChildren, err := b.getChildren(ctx, childTx, db)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		children = append(children, newChildren...)
+	}
+
+	return maxBlock, tx, children, nil
+}
+
+func (b *BlockStorage) getChildren(
+	ctx context.Context,
+	tx *types.Transaction,
+	db database.Transaction,
+) ([]*types.TransactionIdentifier, error) {
+	var children []*types.TransactionIdentifier
+	for _, relatedTx := range tx.RelatedTransactions {
+		if relatedTx.Direction == types.Forward {
+			children = append(children, relatedTx.TransactionIdentifier)
+		}
+	}
+
+	_, err := db.Scan(
+		ctx,
+		getBackwardRelationKey(tx.TransactionIdentifier, nil),
```

Comment:
This is a lot cleaner than maintaining a separate prefix function. We should probably go back and remove any `...prefix()` functions in another PR.